### PR TITLE
Trunk-5710:Implement the getByUniqueId(String uniqueId) in SchedulerS…

### PR DESCRIPTION
**Pull request by ODORA0**
…ervice.java (#3132)

(cherry picked from commit 924526ea581ab613e709e3c078d06c166aacd124)

**Issue worked on**: https://issues.openmrs.org/browse/TRUNK-5710

### CODE REVIEW

**Comment by ODORA0** [comment]
@dkayiwa have a look at this, please.

File: api/src/test/java/org/openmrs/scheduler/SchedulerServiceTest.java [hunk 1]
```diff
@@ -413,4 +414,22 @@ public void shouldSaveLastExecutionTime() throws Exception {
...
+		assertNotNull(td);
+	}
+
+	/**
+	 * @see org.openmrs.scheduler.SchedulerService#getTaskByUuid(java.lang.String)
+	 */
+	@Test
+	public void getTaskByUuid_shouldReturnNullWhenUuidDoesNotExist() throws Exception {
+		TaskDefinition td = Context.getSchedulerService().getTaskByUuid("kncsjvcjvbevismcvbsnksndcsjbvjhvbn");
+		assertNull(td);
```
Comment (jnsereko) [inline api/src/test/java/org/openmrs/scheduler/SchedulerServiceTest.java:433]: Thanks for the good work done
Why do you use a declaration, if it were me i would use the code below.
`assertNull(Context.getSchedulerService().getTaskByUuid("kncsjvcjvbevismcvbsnksndcsjbvjhvbn");`

**Comment by jnsereko** [comment]
May you please share the ticket link.

**Comment by ODORA0** [comment]
> …ervice.java (#3132)
> 
> (cherry picked from commit [924526e](https://github.com/openmrs/openmrs-core/commit/924526ea581ab613e709e3c078d06c166aacd124))

@jnsereko as you may have noticed, this is a cherry pick to be backported to the 2.3.x branch. 
See this https://github.com/openmrs/openmrs-core/pull/3132

**Comment by dkayiwa** [comment]
Can you include a link to the JIRA ticket as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by ODORA0** [comment]
@dkayiwa Sure, just added that in.

File: api/src/main/java/org/openmrs/scheduler/SchedulerService.java [hunk 2]
```diff
@@ -103,6 +103,16 @@
 	@Authorized( { "Manage Scheduler" })
 	public TaskDefinition getTask(Integer id);
 	
+	/**
+	 * @since 2.4.0
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/scheduler/SchedulerService.java:107]: This should be `since 2.3.1, 2.4.0`

**Comment by ODORA0** [comment]
@dkayiwa that's done

**Comment by dkayiwa** [comment]
@ODORA0 can you remind me as to why you want to back port this?

**Comment by ODORA0** [comment]
The `TaskServiceWrapper` service class that is relied on to complete [RESTWS-764](https://issues.openmrs.org/browse/RESTWS-764) doesn't have` getByUniqueId(String uniqueId)` method but instead has `getTaskById(Integer id)` which takes in an integer and not a string uuid

See [THIS](https://issues.openmrs.org/browse/RESTWS-764?focusedCommentId=262025&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-262025)

**Comment by dkayiwa** [comment]
That method is already added and will be available in the next release. So why back port it?

**Comment by ODORA0** [comment]
@dkayiwa see [HERE](https://issues.openmrs.org/browse/TRUNK-5710?focusedCommentId=262918&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-262918)

**Comment by dkayiwa** [comment]
But core 2.3.0 is already released without this change.

**Comment by ODORA0** [comment]
@dkayiwa then that wouldn't make the implementation of the service method in webservicesrest possible that depends on core 2.3, thoughts?

cc @ssmusoke @gitcliff

**Comment by gitcliff** [comment]
@dkayiwa then would creating sub module version of rest module `webservices 2.4` depending on core `2.4.0-SNAPSHOT` be the ideal thing to do in this case ?

**Comment by dkayiwa** [comment]
Correct. And it is already created here https://github.com/openmrs/openmrs-module-webservices.rest/pull/430

**Comment by ODORA0** [comment]
@dkayiwa I hope we can have that PR merged to move forward.

**Comment by dkayiwa** [comment]
@ODORA0 did you understand @gitcliff's response?

**Comment by ODORA0** [comment]
@dkayiwa I was refering to https://github.com/openmrs/openmrs-module-webservices.rest/pull/430

**Comment by ODORA0** [comment]
@dkayiwa Can't we have this cherry pick still backported and that way, @ssmusoke can have a 2.3.1 version released to cater for the UgandEMR that still runs on 2.3?

**Comment by ODORA0** [comment]
@dkayiwa There seems to be an error with the merge of https://github.com/openmrs/openmrs-module-webservices.rest/pull/430 look [HERE](https://hastebin.com/kijawojose.makefile)

**Pull request closed without merging**